### PR TITLE
feat: add CTA base models to Rewired Spoke dbt [DE-317]

### PR DESCRIPTION
**Pull request by emily-flambe**
@kanelouise @jitoquinto I had some "type 2 fun" this weekend writing a script to generate these dbt files based on tables the Rewired Spoke DAG was delivering to BQ... kind of an early attempt at a dbt code generation script (not at all ready for primetime lol).

Anyway these are the dbt models that the script spun out, and they seem to be doing the thing (it's running in prod right now) 👀 

~....Actually I am marking this as a "draft" again, I need to go back and implement incremental base models for the huge tables in this sync before it's super ready for review!~

Incremental models have been created! All seems to be running smoothly in prod composer 👀

### CODE REVIEW

**Comment by emily-flambe** [comment]
@jitoquinto can I get your eyes and (hopefully) approval on this when you have a sec? 👀 (not urgent, just bumping to get it closed out soon)

File: dbt-cta/spoke/models/1_cta_full_refresh/assignable_campaign_contacts_base.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    
+        {{config(
+            cluster_by="_cta_sync_datetime_utc",
+            partition_by={"field": "_cta_sync_datetime_utc", "data_type": "timestamp", "granularity": "day"},
+            partitions=partitions_to_replace,
+            unique_key="id"
+        )}}
+    
+        -- Final base SQL model
+        SELECT
```
[the remark this answers is not in the document]
Comment (jitoquinto) [reply dbt-cta/spoke/models/1_cta_full_refresh/assignable_campaign_contacts_base.sql:14]: i guess we could also just run a linter on these files, did we ever find a good one to use for dbt??

**Review by jitoquinto — DISMISSED** [review]
Left some questions/comments but nothing that would block this!

**Comment by jitoquinto** [comment]
> @jitoquinto can I get your eyes and (hopefully) approval on this when you have a sec? 👀 (not urgent, just bumping to get it closed out soon)

thanks for re-pinging theres soo many pr emails... maybe gotta rethink how many messages i get from github

File: dbt-cta/spoke/models/1_cta_full_refresh/assignable_campaign_contacts_base.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    CAST(`message_status` AS STRING) AS `message_status`,
+    CAST(`texting_hours_end` AS INTEGER) AS `texting_hours_end`,
+        FORMAT("%x", FARM_FINGERPRINT(CONCAT(`_cta_sync_rowid`,
+                                        `_cta_sync_datetime_utc`,
+                                        `campaign_id`,
+                                        `contact_timezone`,
+                                        `id`,
+                                        `message_status`,
+                                        `texting_hours_end`))) AS _unique_row_id
+    FROM {{ source('cta', 'assignable_campaign_contacts_raw') }}
```
[the remark this answers is not in the document]
Comment (emily-flambe) [reply dbt-cta/spoke/models/1_cta_full_refresh/assignable_campaign_contacts_base.sql:29]: correct, they are created from just running queries on tables in postgres and then copying the data over to GCS! So none of the usual flattening shenanigans apply here.

File: dbt-cta/spoke/models/1_cta_full_refresh/assignable_campaign_contacts_base.sql [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    CAST(`id` AS INTEGER) AS `id`,
+    CAST(`message_status` AS STRING) AS `message_status`,
+    CAST(`texting_hours_end` AS INTEGER) AS `texting_hours_end`,
+        FORMAT("%x", FARM_FINGERPRINT(CONCAT(`_cta_sync_rowid`,
+                                        `_cta_sync_datetime_utc`,
+                                        `campaign_id`,
+                                        `contact_timezone`,
+                                        `id`,
+                                        `message_status`,
+                                        `texting_hours_end`))) AS _unique_row_id
```
[the remark this answers is not in the document]
Comment (emily-flambe) [reply dbt-cta/spoke/models/1_cta_full_refresh/assignable_campaign_contacts_base.sql:28]: Yeah, that would be clearer - the queries on the postgres tables create uuids in fields named _cta_sync_rowid, so how about we name this one `_cta_hashid`? Possibly it's confusing to have both of them, but the rowid is just a UUID, whereas the hashid should in theory represent the set of fields that identifies a unique row 🤔 (kinda like _airbye_ab_id vs the hashids created in _ab3).

**Review by jitoquinto — APPROVED** [review]
lgtm!

File: utils/generate_dbt/Pipfile
[the remark this answers is not in the document]
Comment (kanelouise) [reply utils/generate_dbt/Pipfile:1]: very excited to see the script that made these things possible.

**Comment by emily-flambe** [comment]
@kanelouise it is here ICYMI!
https://github.com/community-tech-alliance/dbt-cta/pull/131

**Pull request merged**